server/statistics/hot_peer.go

```diff
@@ -41,15 +78,19 @@ type HotPeerStat struct {
 	KeyRate  float64  `json:"flow_keys"`
 
 	// rolling statistics, recording some recently added records.
-	rollingByteRate *movingaverage.TimeMedian
-	rollingKeyRate  *movingaverage.TimeMedian
+	rollingByteRate *dimStat
+	rollingKeyRate  *dimStat
 
 	// LastUpdateTime used to calculate average write
 	LastUpdateTime time.Time `json:"last_update_time"`
 
-	needDelete bool
-	isLeader   bool
-	isNew      bool
+	needDelete         bool
+	isLeader           bool
+	isNew              bool
+	justTransferLeader bool
+	interval           uint64
+	thresholds         [dimLen]float64
+	peers              []uint64
```

Comment:
these fields are actually used as parameters, better use another struct to wrap.

Reply:
yep, but it will change many functions, could we do it in another pr?